feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go

```diff
@@ -537,18 +587,286 @@ func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, good, bad, lb []
+
+	t.Logf("Adding NH %d for trunk4 via gribi", nh3ID)
+	if deviations.GRIBIMACOverrideWithStaticARP(args.dut) {
+		args.client.AddNH(t, nh3ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk4, Mac: port4mac, Dest: atePort4.IPv4})
+	} else {
+		args.client.AddNH(t, nh3ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk4, Mac: port4mac})
+	}
+
+	t.Log("Validate traffic switching from  ate port2, ate port3 to ate port4")
+	validateTrafficFlows(t, args.ate, []*ondatra.Flow{p4flow}, []*ondatra.Flow{p2flow, p3flow}, nil, 2, args.client, true)
+
+	t.Log("Validate traffic switching from  ate port4 back to ate port2 and ate port3")
+	validateTrafficFlows(t, args.ate, nil, []*ondatra.Flow{p4flow}, []*ondatra.Flow{p2flow, p3flow}, 3, args.client, true)
+
+}
+
+// deleteinterfaceconfig unconfigs interfaces.
+func deleteinterfaceconfig(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+	p2 := dut.Port(t, "port2")
+	p3 := dut.Port(t, "port3")
+	p4 := dut.Port(t, "port4")
+
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		ni := deviations.DefaultNetworkInstance(dut)
+		gnmi.Delete(t, dut, d.NetworkInstance(ni).Interface(p2.Name()+".").Subinterface().Config())
+		gnmi.Delete(t, dut, d.NetworkInstance(ni).Interface(p3.Name()+".").Subinterface().Config())
+		gnmi.Delete(t, dut, d.NetworkInstance(ni).Interface(p4.Name()+".").Subinterface().Config())
+	} else {
+		gnmi.Delete(t, dut, d.Interface(p2.Name()).Subinterface(0).Config())
+		gnmi.Delete(t, dut, d.Interface(p3.Name()).Subinterface(0).Config())
+		gnmi.Delete(t, dut, d.Interface(p4.Name()).Subinterface(0).Config())
+	}
+}
+
+// configDUTDrain configures ports for drain test.
+func configDUTDrain(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+	p2 := dut.Port(t, "port2")
+	p3 := dut.Port(t, "port3")
+	p4 := dut.Port(t, "port4")
+	btrunk2 = netutil.NextAggregateInterface(t, dut)
+
+	i2 := &oc.Interface{Name: ygot.String(btrunk2)}
+	gnmi.Replace(t, dut, d.Interface(*i2.Name).Config(), configInterfaceDUT(i2, &dutPort2, dut))
+	T2 := configureBundle(t, p2.Name(), *i2.Name)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(p2.Name()).Config(), T2)
+
+	btrunk3 = netutil.NextAggregateInterface(t, dut)
+	i3 := &oc.Interface{Name: ygot.String(btrunk3)}
+	gnmi.Replace(t, dut, d.Interface(*i3.Name).Config(), configInterfaceDUT(i3, &dutPort3, dut))
+	T3 := configureBundle(t, p3.Name(), *i3.Name)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(p3.Name()).Config(), T3)
+
+	btrunk4 = netutil.NextAggregateInterface(t, dut)
+	i4 := &oc.Interface{Name: ygot.String(btrunk4)}
+	gnmi.Replace(t, dut, d.Interface(*i4.Name).Config(), configInterfaceDUT(i4, &dutPort4, dut))
+	T4 := configureBundle(t, p4.Name(), *i4.Name)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(p4.Name()).Config(), T4)
+
```

Comment:
Please could you add code to enable ipv6 enable for AE interfaces like it is enabled in function "configureDUT" for p2/p3/p4.


	if deviations.ExplicitIPv6EnableForGRIBI(dut) {
		gnmi.Update(t, dut, d.Interface(*i2.Name).Subinterface(0).Ipv6().Enabled().Config(), true)
		gnmi.Update(t, dut, d.Interface(*i3.Name).Subinterface(0).Ipv6().Enabled().Config(), true)
		gnmi.Update(t, dut, d.Interface(*i4.Name).Subinterface(0).Ipv6().Enabled().Config(), true)
	}

Reply:
Have added this now. But this is for testImplementDrain (case2) which runs only for Cisco now as static route/arp deviations needed for other vendors.